### PR TITLE
Add MNIST starter

### PR DESCRIPTION
**Pull request by markwinter**
Adding a simple mnist example. Added a brief tutorial in the REAMDE that users can follow

I changed the example from the pytorch one to:
- Only train 1 epoch, this is good enough for a demo
- Save model file by default, and save it to `./uwstore/output/model_cnn.pt`

### CODE REVIEW

File: mnist/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+## Run the starter on Unweave
+
+1. If you don't have Unweave CLI installed, you can get it from homebrew
+
+```bash
+$ brew install unweave/unweave/unweave
+==> Downloading https://github.com/unweave/cli/releases/download/v0.0.19/unweave_0.0.19_darwin_arm64.tar.gz
+Already downloaded: /Users/markwinter/Library/Caches/Homebrew/downloads/fb4b2b676e3b8680405f5094e438827fead5210146df54cc96aad0ea026ae1e3--unweave_0.0.19_darwin_arm64.tar.gz
+==> Installing unweave from unweave/unweave
+🍺  /opt/homebrew/Cellar/unweave/0.0.19: 5 files, 7.4MB, built in 1 second
```
Comment (noorvir) [inline mnist/README.md:16]: I don't think we need to have the terminal output here. Might be confusing for people since the unweave version might be different from them. Also we should try and minimize unnecessary information in the docs i.e. minimize cognitive load

File: mnist/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+```bash
+$ unweave init
+? Enter project name [mnist]: mnist-starter
+Created project mnist-starter
+Path:    	/Users/markwinter/Code/starters/mnist
+Project: 	mnist-starter
+```
+
+5. Run the training script with the Unweave CLI. This will create a zepl with your data and run it on our infrastructure.
```
Comment (noorvir) [inline mnist/README.md:43]: Could you change this to: "This will create a serverless compute node (`zepl`) with your data and run it."

File: mnist/main.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        test_kwargs.update(cuda_kwargs)
+
+    transform=transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize((0.1307,), (0.3081,))
+        ])
+    dataset1 = datasets.MNIST('../data', train=True, download=True,
+                       transform=transform)
+    dataset2 = datasets.MNIST('../data', train=False,
+                       transform=transform)
```
Comment (noorvir) [inline mnist/main.py:120]: We want this path to be in the `uwstore/data` directory.

File: mnist/README.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+🔄 Syncing Unweave Store
+
+🔄 Building environment
+
+🚀 Zepl running. Fetching logs ...
+
+Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz
+Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz to ../data/MNIST/raw/train-images-idx3-ubyte.gz
+100.0%
+Extracting ../data/MNIST/raw/train-images-idx3-ubyte.gz to ../data/MNIST/raw
```
Comment (noorvir) [inline mnist/README.md:67]: This should download to the `./uwstore/data` directory. In the future, we should have a flow that automatically adds data to the user's `uwstore` _before_ the run the script. For now this is good though.

**Review by noorvir — CHANGES_REQUESTED** [review]

**Review by noorvir — APPROVED** [review]
